controllers/machinedeletion_controller.go

```diff
@@ -95,30 +140,54 @@ func (r *MachineDeletionReconciler) getRemediation(ctx context.Context, req ctrl
 	return remediation
 }
 
-func (r *MachineDeletionReconciler) deleteMachine(ctx context.Context, machine *unstructured.Unstructured) error {
-	if err := r.Client.Delete(ctx, machine); err != nil {
-		if !errors.IsNotFound(err) {
-			r.Log.Error(err, "error deleting machine %s in namespace %s: %v", machine.GetName(), machine.GetNamespace())
-			return err
-		}
-		r.Log.Info("machine: %s in namespace: %s is deleted , but remediation still exist", machine.GetName(), machine.GetNamespace())
+func (r *MachineDeletionReconciler) getNodeFromMdr(mdr *v1alpha1.MachineDeletion) (*v1.Node, error) {
+	node := &v1.Node{}
+	key := client.ObjectKey{
+		Name: mdr.Name,
 	}
-	return nil
-}
 
-func getMachineOwnerRef(remediation *v1alpha1.MachineDeletion) *v1.OwnerReference {
-	for _, ownerRef := range remediation.OwnerReferences {
-		if ownerRef.Kind == machineKind {
-			return &ownerRef
-		}
+	if err := r.Get(context.TODO(), key, node); err != nil {
+		return nil, err
 	}
-	return nil
+	return node, nil
 }
-func buildMachine(ref *v1.OwnerReference, remediation *v1alpha1.MachineDeletion) *unstructured.Unstructured {
+
+func (r *MachineDeletionReconciler) buildMachineFromNode(node *v1.Node) (*unstructured.Unstructured, error) {
+
+	var nodeAnnotations map[string]string
+	if nodeAnnotations = node.Annotations; nodeAnnotations == nil {
+		return nil, fmt.Errorf(noAnnotationsError, node.Name)
+	}
+	var machineNameNamespace, machineName string
+
+	//OpenShift Machine
+	if machineNameNamespace = nodeAnnotations[machineAnnotationOpenshift]; len(machineNameNamespace) == 0 {
+		return nil, fmt.Errorf(noMachineAnnotationError, node.Name)
```

Comment:
this error will be printed even if the annotation exists but with empty value.
consider using:
`if machineNameNamespace, exists = nodeAnnotations[machineAnnotationOpenshift]; !exists {`